weechat: build with python3 by default

Ma27 opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

WeeChat also supports Python3 for scripts which should be preferred as
CPython2 is about to get EOLed soon: https://weechat.org/scripts/python3/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

Ma27 committed:
weechat: build with python3 by default

WeeChat also supports Python3 for scripts which should be preferred as
CPython2 is about to get EOLed soon: https://weechat.org/scripts/python3/

jonringer commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21205,11 +21205,14 @@ in
   weechat = wrapWeechat weechat-unwrapped { };
 
-  weechatScripts = dontRecurseIntoAttrs (callPackage ../applications/networking/irc/weechat/scripts { });
+  weechatScripts = dontRecurseIntoAttrs (callPackage ../applications/networking/irc/weechat/scripts {
```
dontRecurseIntoAttrs is now the default, I think the implementation now is just the id function `drv: drv`
```suggestion
  weechatScripts =  callPackage ../applications/networking/irc/weechat/scripts {
```

jonringer replied:
yep, had to check https://github.com/NixOS/nixpkgs/blob/5e86891baa57ba4e8aace81c0f24399df3082d37/pkgs/top-level/all-packages.nix#L75

Ma27 replied:
In fact it's always been the default, `dontRecurseIntoAttrs` mainly serves a documentation purpose. To quote the comment above the function:

```
  # This is intended to be the reverse of recurseIntoAttrs, as it is
  # defined now it exists mainly for documentation purposes, but you
  # can also override this with recurseIntoAttrs to recurseInto all
  # the Attrs which is useful for testing massive changes. Ideally,
  # every package subset not marked with recurseIntoAttrs should be
  # marked with this.
```

jonringer replied:
ah, if it has a purpose, then I'm fine with leaving it in :)

Mic92 commented:
I agree we should have that at some point. The only thing that concerns me a bit that even archlinux, who were early python3 adopters, build weechat with python2: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/weechat#n13
On the other hand it seems that most of the python2-only scripts seems quite old: https://weechat.org/scripts/stable/language/python2-only/
Personally I don't use any of plugins, but would be interested how other weechat users think about this.

Ma27 commented:
> I agree we should have that at some point. The only thing that concerns me a bit that even archlinux, who were early python3 adopters, build weechat with python2

Last time I checked (about a month ago) they were building a lot of applications with python2, but I'm no arch user, so could've been a false impression.

> On the other hand it seems that most of the python2-only scripts seems quite old: https://weechat.org/scripts/stable/language/python2-only/

After taking a look at the list, I didn't find any of our currently supported plugins in there. However I guess I should check all of the python stuff manually (just checked `wee-slack` and looked if any of our plugins require python2).

Ma27 commented:
So, short update: rechecked all scripts and it seems as `weechat-xmpp` requires a `xmpppy` with Python3 support, currently investigating if that's possible.

After that we'd need somebody with more darwin experience to fix the build, after that this should be ready IMHO.

Ma27 committed:
weechatScripts.weechat-xmpp: remove

This plugin is fairly outdated and depends on python2 libraries that
don't receive any updates either (xmpppy for instance[1]).

[1] https://pypi.org/project/xmpppy/

Ma27 commented:
Rebased onto current master, also dropped weechat-xmpp as it relies on pretty old dependencies and appears rather unmaintained atm.

jonringer commented:
There's also a reference to the weechat-xmpp in aliases
```
attribute 'weechat-xmpp' missing, at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-2-shlevy.ewr1.nix.ci/pkgs/top-level/aliases.nix:383:18
```

Ma27 commented:
Ahh, right, fixed! :)

Mic92 commented on pkgs/top-level/aliases.nix:
```diff
@@ -380,7 +380,6 @@ mapAliases ({
   virtviewer = virt-viewer; # added 2015-12-24
   vorbisTools = vorbis-tools; # added 2016-01-26
   webkit = webkitgtk; # added 2019-03-05
-  weechat-xmpp = weechatScripts.weechat-xmpp; # added 2018-09-06
```
Maybe?

```
weechat-xmpp = throw "weechat-xmpp has been removed as it doesn't receive
     any updates from upstream and depends on outdated Python2-based modules.";
```

lheckemann replied:
Not in favour. Removal already throws an error, and the reason for removal is documented in the release notes.
